packages/trader/src/Modules/MT5/Containers/mt5-password-manager-modal.jsx

```diff
@@ -131,125 +130,209 @@ const MT5PasswordSuccessMessage = ({ toggleModal, is_investor }) => (
     </div>
 );
 
-class MT5PasswordManagerModal extends React.Component {
-    multi_step_ref = React.createRef();
-    state = {
-        active_tab_index: 0,
-        error_message_main: '',
-        is_submit_success_main: false,
-        error_message_investor: '',
-        is_submit_success_investor: false,
+const MT5PasswordManagerModal = ({
+    enableApp,
+    email,
+    disableApp,
+    is_visible,
+    selected_login,
+    selected_account,
+    toggleModal,
+    selected_account_type,
+    selected_account_group,
+    sendVerifyEmail,
+}) => {
+    const [active_tab_index, setActiveTabIndex] = React.useState(0);
+    const [main_error_message, setMainErrorMessage] = React.useState('');
+    const [is_main_submit_success, setMainSubmitSuccess] = React.useState(false);
+    const [investor_error_message, setInvestorErrorMessage] = React.useState('');
+    const [is_investor_submit_success, setInvestorSubmitSuccess] = React.useState(false);
+    const [password_type, setPasswordType] = React.useState('');
+
+    const prev_is_visible = usePreviousState(is_visible);
+    const multi_step_ref = React.useRef();
+
+    React.useEffect(() => {
+        if (!prev_is_visible && is_visible) {
+            reset(0);
+        }
+    }, [is_visible, prev_is_visible]);
```

Comment:
I guess we don't need `prev_is_visible`. 
if `is_visible` is `true`, then the previous value will be `false` or `undefined`. We can just check if `is_visible` is true at `L156`